### PR TITLE
Rel Notes for Storage 4.14

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.14 only

Issue: release note 4.14 storage changes

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

**PTAL**: @dobsonj, @tsmetana, @gnufied, @jsafrane, @bertinatto, @gcharot, @ropatil010, @chao007, @radeore, @duanwei33

### CODE REVIEW

File: release_notes/ocp-4-14-release-notes.adoc [hunk 1]
```diff
@@ -170,6 +170,52 @@ With this release, {product-title} updated to HAProxy 2.6.
...
+
+This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users.
+
+Although storage class referencing to the in-tree storage plug-in will continue working, it is recommended that you switch the default storage class to the CSI storage class.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
+
+[id="ocp-4-14-storage-secrets-store-csi-driver-operator]
+==== Secrets Store CSI Driver Operator (Technology Preview)
+The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows [product-title] to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status.
```
Comment (skrthomas) [inline release_notes/ocp-4-14-release-notes.adoc:209]:
```suggestion
The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status.
```

File: release_notes/ocp-4-14-release-notes.adoc [hunk 2]
```diff
@@ -170,6 +170,52 @@ With this release, {product-title} updated to HAProxy 2.6.
...
+
+[id="ocp-4-14-storage-automigration-vsphere"]
+==== Automatic CSI migration for VMware vSphere
+Starting with {product-title} 4.8, automatic migration for in-tree volume plugins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. 
+
+Support for vSphere was provided in this feature in {product-title} 4.10 at the Technology Preview level, and vSphere support became generally available in 4.13. However, while for new installations of {product-title}, automatic migration was enabled by default, when upgrading from {product-title} 4.12, or earlier, to 4.13, automatic CSI migration for vSphere only occurred if you opted in. In {product-title} 4.14, CSI migration for vSphere is now enabled by default under all circumstances and requires no action by an administrator.
+
+This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users.
+
+Although storage class referencing to the in-tree storage plug-in will continue working, it is recommended that you switch the default storage class to the CSI storage class.
```
Comment (skrthomas) [inline release_notes/ocp-4-14-release-notes.adoc:203]:
```suggestion
Although storage class referencing to the in-tree storage plug-in continues working, it is recommended that you switch the default storage class to the CSI storage class.
```

File: release_notes/ocp-4-14-release-notes.adoc [hunk 3]
```diff
@@ -170,6 +170,52 @@ With this release, {product-title} updated to HAProxy 2.6.
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
+
+[id="ocp-4-14-storage-secrets-store-csi-driver-operator]
+==== Secrets Store CSI Driver Operator (Technology Preview)
+The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows [product-title] to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status.
+
+For more information, see ???add link for storage content & authentication content???
```
Comment (skrthomas) [inline release_notes/ocp-4-14-release-notes.adoc:211]: Don't forget to add your link here :)

File: release_notes/ocp-4-14-release-notes.adoc [hunk 4]
```diff
@@ -170,6 +170,52 @@ With this release, {product-title} updated to HAProxy 2.6.
...
+
+[id="ocp-4-14-storage-azure-file-nfs]
+==== Azure File supporting NFS is generally available
+{product-title} 4.14 supports Azure File Container Storage Interface (CSI) Driver Operator with Network File System (NFS) as generally available.
+
+For more information, see ???.
```
Comment (skrthomas) [inline release_notes/ocp-4-14-release-notes.adoc:217]: link?

**Review by skrthomas — COMMENTED** [review]
A couple of comments and wondering about missing links. I guess once the linked docs are merged, this build won't be broken?

File: release_notes/ocp-4-14-release-notes.adoc [hunk 5]
```diff
@@ -175,6 +175,52 @@ With this release, you can access the NMstate Operator and resources such as the
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-google-cloud-file.adoc[Google Compute Platform Filestore CSI Driver Operator].
+
+[id="ocp-4-14-storage-automigration-vsphere"]
+==== Automatic CSI migration for VMware vSphere
+Starting with {product-title} 4.8, automatic migration for in-tree volume plugins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. 
+
+Support for vSphere was provided in this feature in {product-title} 4.10 at the Technology Preview level, and vSphere support became generally available in 4.13. However, while for new installations of {product-title}, automatic migration was enabled by default, when upgrading from {product-title} 4.12, or earlier, to 4.13, automatic CSI migration for vSphere only occurred if you opted in. In {product-title} 4.14, CSI migration for vSphere is now enabled by default under all circumstances and requires no action by an administrator.
+
+This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users.
```
Comment (gcharot) [inline release_notes/ocp-4-14-release-notes.adoc:206]: Looks good from a general standpoint. We may have to link to a KCS  that describes the two remaining issues

File: release_notes/ocp-4-14-release-notes.adoc [hunk 6]
```diff
@@ -175,6 +175,52 @@ With this release, you can access the NMstate Operator and resources such as the
...
 === Storage
 
+[id="ocp-4-14-storage-standardized-sts-config"]
+==== Standardized STS configuration workflow
+{product-title} 4.14 provides a streamlined and standardized procedure to configure Security Token Service (STS) with the AWS Elastic File Storage (EFS) Container Storage Interface (CSI) Driver Operator.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc#efs-sts_persistent-storage-csi-aws-efs[Configuring AWS EFS CSI Driver Operator with Security Token Service].
+
+[id="ocp-4-14-storage-rwod-access-mode"]
+==== Read Write Only Pod access mode (Technology Preview)
```
Comment (radeore) [inline release_notes/ocp-4-14-release-notes.adoc:185]: Can we please update "Read Write Only Pod" to "Read Write Once Pod"

File: release_notes/ocp-4-14-release-notes.adoc [hunk 7]
```diff
@@ -175,6 +175,52 @@ With this release, you can access the NMstate Operator and resources such as the
...
+{product-title} 4.14 introduces a new access mode for persistent volumes (PVs) and persistent volume claims (PVCs) called ReadWriteOncePod (RWOD), which can be used only in a single pod on a single node. This is compared to the existing ReadWriteOnce access mode where a PV or PVC can be used on a single node by many pods. This feature is supported with Technology Preview status.
```
Comment (radeore) [inline release_notes/ocp-4-14-release-notes.adoc:186]: Can we please update "RWOD" to "RWOP"

File: release_notes/ocp-4-14-release-notes.adoc [hunk 8]
```diff
@@ -175,6 +175,52 @@ With this release, you can access the NMstate Operator and resources such as the
...
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
+
+[id="ocp-4-14-storage-secrets-store-csi-driver-operator"]
+==== Secrets Store CSI Driver Operator (Technology Preview)
+The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status.
+
+For more information, see ???add link for storage content & authentication content???
+
+[id="ocp-4-14-storage-azure-file-nfs"]
+==== Azure File supporting NFS is generally available
```
Comment (gcharot) [inline release_notes/ocp-4-14-release-notes.adoc:219]: @lpettyjo 
Azure File CSI NFS support is generally available?

File: release_notes/ocp-4-14-release-notes.adoc [hunk 9]
```diff
@@ -175,6 +175,52 @@ With this release, you can access the NMstate Operator and resources such as the
...
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc#efs-sts_persistent-storage-csi-aws-efs[Configuring AWS EFS CSI Driver Operator with Security Token Service].
+
+[id="ocp-4-14-storage-rwod-access-mode"]
+==== Read Write Only Pod access mode (Technology Preview)
+{product-title} 4.14 introduces a new access mode for persistent volumes (PVs) and persistent volume claims (PVCs) called ReadWriteOncePod (RWOD), which can be used only in a single pod on a single node. This is compared to the existing ReadWriteOnce access mode where a PV or PVC can be used on a single node by many pods. This feature is supported with Technology Preview status.
+
+For more information, see xref:../storage/understanding-persistent-storage.adoc#pv-access-modes_understanding-persistent-storage[Access modes].
+
+[id="ocp-4-14-storage-robust-volmgr"]
+==== Robust VolumeManager reconstruction (Technology Preview)
```
Comment (radeore) [inline release_notes/ocp-4-14-release-notes.adoc:191]: As per the last conversation, this story was moved to 4.15, in that case can we please remove documentation for this story? @jsafrane Can you please confirm?

Comment (gcharot) [inline release_notes/ocp-4-14-release-notes.adoc:191]: @lpettyjo 
This one slipped to 4.15. 
cc @jsafrane

File: release_notes/ocp-4-14-release-notes.adoc [hunk 10]
```diff
@@ -811,6 +857,21 @@ In the following tables, features are marked with the following statuses:
 |Technology Preview
 |Technology Preview
 
+|NFS support for Azure File CSI Operator Driver
+|Not Available
+|Not Available
+|Generally Available
+
+|Read Write Only Pod access mode
```
Comment (radeore) [inline release_notes/ocp-4-14-release-notes.adoc:865]: Can we please update "Read Write Only Pod" to "Read Write Once Pod" ?

File: release_notes/ocp-4-14-release-notes.adoc (hunk 6 above)
Comment (gcharot) [reply release_notes/ocp-4-14-release-notes.adoc:185]: +1 @lpettyjo

File: release_notes/ocp-4-14-release-notes.adoc (hunk 7 above)
Comment (gcharot) [reply release_notes/ocp-4-14-release-notes.adoc:186]: +1 @lpettyjo

**Comment by duanwei33** [comment]
Is it worth mentioning Azure File / Azure Disk support for managed identities?
If it is mentioned in installer part, I think it's okay to omit here.

**Comment by duanwei33** [comment]
In "Deprecated and removed features" section 
"vSphere 7.0 Update 1 or earlier" should be "Deprecated"(instead of "General Availability") in 4.12. I believe it belongs to installer part, so I leave [comments in their jira](https://issues.redhat.com/browse/OSDOCS-6729?focusedId=22837656&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22837656).

File: release_notes/ocp-4-14-release-notes.adoc [hunk 11]
```diff
@@ -175,6 +175,52 @@ With this release, you can access the NMstate Operator and resources such as the
...
+
+For more information, see xref:../storage/understanding-persistent-storage.adoc#pv-access-modes_understanding-persistent-storage[Access modes].
+
+[id="ocp-4-14-storage-robust-volmgr"]
+==== Robust VolumeManager reconstruction (Technology Preview)
+This feature changes how the kubelet starts and cleans up orphaned volumes. This change improves how the node and kubelet startup. If you find any issues with this feature, please report them. This feature is supported with Technology Preview status.
+
+[id="ocp-4-14-storage-gcp-filestore-csi-driver"]
+==== GCP Filestore storage CSI Driver Operator is generally available
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Google Compute Platform (GCP) Filestore Storage. The GCP Filestore CSI Driver Operator that manages this driver is now generally available.
```
[the remark this answers is not in the document]
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:196]: Added it.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 5 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:206]: OK. Keeping this comment open for this.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 6 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:185]: Fixed!

File: release_notes/ocp-4-14-release-notes.adoc (hunk 7 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:186]: Done.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 10 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:884]: Done.

**Comment by lpettyjo** [comment]
> vSphere 7.0 Update 1 or earlier

@duanwei33 I've corrected this.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 9 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:191]: Done.

Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:191]: Removed.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 8 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:215]: @gcharot agreed this is GA, but that there is a known bug that needs to be documented.

**Comment by lpettyjo** [comment]
Need to add known bug for NFS Azure File support.

File: release_notes/ocp-4-14-release-notes.adoc [hunk 12]
```diff
@@ -256,6 +256,50 @@ For more information, see xref:../networking/k8s_nmstate/k8s-nmstate-updating-no
...
+For more information, see xref:../storage/understanding-persistent-storage.adoc#pv-access-modes_understanding-persistent-storage[Access modes].
+
+[id="ocp-4-14-storage-gcp-filestore-csi-driver"]
+==== GCP Filestore storage CSI Driver Operator is generally available
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Google Compute Platform (GCP) Filestore Storage. The GCP Filestore CSI Driver Operator was introduced in {product-title} 4.12 with Technology Preview support. The GCP Filestore CSI Driver Operator is now generally available.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-google-cloud-file.adoc[Google Compute Platform Filestore CSI Driver Operator].
+
+[id="ocp-4-14-storage-automigration-vsphere"]
+==== Automatic CSI migration for VMware vSphere
```
Comment (gcharot) [inline release_notes/ocp-4-14-release-notes.adoc:278]: @lpettyjo We need to add the vsphere version and OCP upgrades consideration here

File: release_notes/ocp-4-14-release-notes.adoc [hunk 13]
```diff
@@ -276,19 +276,60 @@ Support for the Broadcom BCM57504 network interface controller is now available
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
+
+[id="ocp-4-14-storage-secrets-store-csi-driver-operator"]
+==== Secrets Store CSI Driver Operator (Technology Preview)
+The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status.
```
Comment (gcharot) [inline release_notes/ocp-4-14-release-notes.adoc:321]: (..) and certificates stored in enterprise-grade external secrets stores into pods as a inline ephemeral volume.

**Comment by dobsonj** [comment]
lgtm

**Comment by jsafrane** [comment]
lgtm

**Comment by tsmetana** [comment]
lgtm (CSI Filestore GA)

**Comment by gnufied** [comment]
lgtm

File: release_notes/ocp-4-14-release-notes.adoc (hunk 12 above)
Comment (lpettyjo) [reply release_notes/ocp-4-14-release-notes.adoc:369]: Done.

**Comment by bertinatto** [comment]
lgtm

**Comment by duanwei33** [comment]
@lpettyjo Hi, did you add the azure-file-nfs known issue [OCPBUGS-18581](https://issues.redhat.com/browse/OCPBUGS-18581) somewhere?

**Review by bergerhoffer — COMMENTED** [review]
@lpettyjo I added a suggestion for the link to my section. My secrets store PR merged, so if you rebase you should get the updates and be able to check that the xref I gave actually works.

**Comment by lpettyjo** [comment]
> @lpettyjo Hi, did you add the azure-file-nfs known issue [OCPBUGS-18581](https://issues.redhat.com/browse/OCPBUGS-18581) somewhere?

@duanwei33 It's not directly in these changes, but I've updated the JIRA card with the RN text and marked it for inclusion when all of the bugs are added.

**Comment by gcharot** [comment]
lgtm

**Comment by ropatil010** [comment]
LGTM, Thanks..!

**Comment by chao007** [comment]
lgtm

**Comment by radeore** [comment]
LGTM

**Comment by duanwei33** [comment]
LGTM

**Pull request merged**